pkg/build/graph/edges.go

```diff
@@ -40,6 +40,9 @@ const (
 func AddBuildEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildConfigNode) {
 	for _, n := range g.(graph.Graph).Nodes() {
 		if buildNode, ok := n.(*buildgraph.BuildNode); ok {
+			if buildNode.Build.Namespace != node.BuildConfig.Namespace {
+				continue
+			}
```

Comment:
In this file in the diff: `imageRefNode`'s the `DockerImage` branch isn't right because the `DockerRepositoryNode` doesn't have sufficient information to make a choice.  Probably need to update `EnsureDockerRepositoryNode`